src/stores/modelStore.ts

```diff
@@ -0,0 +1,141 @@
+  tags: Ref<string[]> = ref([])
+  /** Metadata: image for the model */
+  image = ref('')
+  /** Whether the model metadata has been loaded from the server, used for `load()` */
+  has_loaded_metadata = ref(false)
+
+  constructor(name: string, directory: string) {
+    this.name.value = name
+    this.title.value = name
+    this.directory.value = directory
+  }
+
+  /** Loads the model metadata from the server, filling in this object if data is available */
+  async load(): Promise<void> {
+    if (this.has_loaded_metadata.value) {
+      return
+    }
+    const metadata = await api.viewMetadata(
+      this.directory.value,
+      this.name.value
+    )
+    if (!metadata) {
+      return
+    }
+    this.title.value =
+      _findInMetadata(
+        metadata,
+        'modelspec.title',
+        'title',
+        'display_name',
+        'name'
+      ) || this.name.value
+    this.architecture_id.value =
+      _findInMetadata(metadata, 'modelspec.architecture', 'architecture') || ''
+    this.author.value =
+      _findInMetadata(metadata, 'modelspec.author', 'author') || ''
+    this.description.value =
+      _findInMetadata(metadata, 'modelspec.description', 'description') || ''
+    this.resolution.value =
+      _findInMetadata(metadata, 'modelspec.resolution', 'resolution') || ''
+    this.usage_hint.value =
+      _findInMetadata(metadata, 'modelspec.usage_hint', 'usage_hint') || ''
+    this.trigger_phrase.value =
+      _findInMetadata(metadata, 'modelspec.trigger_phrase', 'trigger_phrase') ||
+      ''
+    this.image.value =
+      _findInMetadata(
+        metadata,
+        'modelspec.thumbnail',
+        'thumbnail',
+        'image',
+        'icon'
+      ) || ''
+    const tagsCommaSeparated =
+      _findInMetadata(metadata, 'modelspec.tags', 'tags') || ''
+    this.tags.value = tagsCommaSeparated.split(',').map((tag) => tag.trim())
+    this.has_loaded_metadata.value = true
+  }
+}
+
+/** Model store for a folder */
+export class ModelStore {
+  models: Record<string, ComfyModelDef> = reactive({})
+
+  constructor(directory: string, models: string[]) {
+    for (const model of models) {
+      this.models[model] = new ComfyModelDef(model, directory)
+    }
+  }
+
+  async loadModelMetadata(modelName: string) {
+    if (this.models[modelName]) {
+      await this.models[modelName].load()
+    }
+  }
+}
+
+/** Model store handler, wraps individual per-folder model stores */
+export const useModelStore = defineStore('modelStore', {
+  state: () => ({
+    modelStoreMap: {} as Record<string, ModelStore>
+  }),
+  actions: {
+    async getModelsInFolderCached(folder: string): Promise<ModelStore> {
+      if (folder in this.modelStoreMap) {
+        return this.modelStoreMap[folder]
+      }
+      // TODO: needs a lock to avoid overlapping calls
+      const models = await api.getModels(folder)
+      if (!models) {
+        return null
+      }
+      const store = reactive(new ModelStore(folder, models))
```

Comment:
Same here. `reactive` can be dropped, as anything stored in Pinia is marked as reactive deeply (The whole recursive object tree are marked as reactive automatically)